Library/Config.ts

```diff
@@ -8,37 +8,64 @@ class Config {
     // This key is provided in the readme
     public static ENV_iKey = "APPINSIGHTS_INSTRUMENTATIONKEY";
     public static legacy_ENV_iKey = "APPINSIGHTS_INSTRUMENTATION_KEY";
+    public static ENV_profileQueryEndpoint = "APPINSIGHTS_PROFILE_QUERY_ENDPOINT";
 
     public instrumentationKey: string;
-    public instrumentationKeyHash: string;
+    public correlationId: string;
     public sessionRenewalMs: number;
     public sessionExpirationMs: number;
+    public endpointBase: string;
     public endpointUrl: string;
     public maxBatchSize: number;
     public maxBatchIntervalMs: number;
     public disableAppInsights: boolean;
     public samplingPercentage: number;
+    public correlationIdRetryIntervalMs: number;
+
+    private setCorrelationId: (string) => void;
+    private _profileQueryEndpoint: string;
 
     // A list of domains for which correlation headers will not be added.
     public correlationHeaderExcludedDomains: string[];
 
     constructor(instrumentationKey?: string) {
         this.instrumentationKey = instrumentationKey || Config._getInstrumentationKey();
-        this.instrumentationKeyHash = Config._getStringHashBase64(this.instrumentationKey);
-        this.endpointUrl = "https://dc.services.visualstudio.com/v2/track";
+        this.endpointBase = "https://dc.services.visualstudio.com";
+        this.endpointUrl = `${this.endpointBase}/v2/track`;
         this.sessionRenewalMs = 30 * 60 * 1000;
         this.sessionExpirationMs = 24 * 60 * 60 * 1000;
         this.maxBatchSize = 250;
         this.maxBatchIntervalMs = 15000;
         this.disableAppInsights = false;
         this.samplingPercentage = 100;
+        this.correlationIdRetryIntervalMs = 30 * 1000;
         this.correlationHeaderExcludedDomains = [
             "*.blob.core.windows.net", 
             "*.blob.core.chinacloudapi.cn",
             "*.blob.core.cloudapi.de",
             "*.blob.core.usgovcloudapi.net"];
+        
+        this.setCorrelationId = (correlationId) => this.correlationId = correlationId;
+
+        this.profileQueryEndpoint = process.env[Config.ENV_profileQueryEndpoint] || this.endpointBase;
     }
 
+    public set profileQueryEndpoint(endpoint: string) {
+        CorrelationIdManager.cancelCorrelationIdQuery(this._profileQueryEndpoint, this.instrumentationKey, this.setCorrelationId);
```

Comment:
Do we need unit test to verify that cancelCorrelationIdQuery is called and that it cancels pending queries?